src/model/channel/channel_id.rs

```diff
@@ -55,21 +55,22 @@ impl ChannelId {
         guild_id: Option<GuildId>,
     ) -> Result<GuildChannel> {
         #[cfg(feature = "cache")]
-        if let Some(cache) = cache_http.cache()
-            && let Some(guild_id) = guild_id
-            && let Some(guild) = cache.guild(guild_id)
-            && let Some(channel) = guild.channels.get(&self)
-        {
-            return Ok(channel.clone());
-        }
-
-        #[cfg(feature = "temp_cache")]
-        if let Some(temp_channel) = cache.temp_channels.get(&self) {
-            if guild_id.is_some_and(|id| temp_channel.base.guild_id != id) {
-                return Err(Error::Model(ModelError::ChannelNotFound));
+        if let Some(cache) = cache_http.cache() {
+            if let Some(guild_id) = guild_id
+                && let Some(guild) = cache.guild(guild_id)
+                && let Some(channel) = guild.channels.get(&self)
+            {
+                return Ok(channel.clone());
```

Comment:
Given the Clippy error in CI, I can see how the mistake happened. I am not sure if it is a fault on Clippy's side, but I would simply silence the error since it is wrong:
```suggestion
        // Ignore clippy, the two `if let`s must be separated
        #[allow(clippy::collapsible_if)]
        if let Some(cache) = cache_http.cache() {
            if let Some(guild_id) = guild_id
                && let Some(guild) = cache.guild(guild_id)
                && let Some(channel) = guild.channels.get(&self)
            {
                return Ok(channel.clone());
```